crypto/cms/cms_sd.c

```diff
@@ -1034,8 +1047,11 @@ int CMS_SignerInfo_verify_content(CMS_SignerInfo *si, BIO *chain)
         if (EVP_PKEY_CTX_set_signature_md(pkctx, md) <= 0)
             goto err;
         si->pctx = pkctx;
-        if (!cms_sd_asn1_ctrl(si, 1))
+        if (!cms_sd_asn1_ctrl(si, 1)) {
+            si->pctx = NULL;
             goto err;
+        }
+        si->pctx = NULL;
```

Comment:
I realize the assignment was already there, but do we care about the prior value of `si->pctx`? Does it need to be freed?

Reply:
I don't think that it can ever happen that `si->pctx` is non-null here.
If that happens, it would probably be a pre-existing bug, and a test case where
that happens would be necessary to know what to do with this pctx.
I think it would be in worst case a small memory leak, but if I free something
that I don't own, we can easily have a crash.

Reply:
Anyway none of the changes in this PR are theoretic issues,
I have all of them reproduced by error injection by having this injection test
run for about 3 months:
https://github.com/bernd-edlinger/openssl/blob/update_111_feature_workflows/doit.sh
Well obviously against my 111 feature branch since 3.x would not withstand such
a test procedure for more than a minute.

Reply:
> I don't think that it can ever happen that si->pctx is non-null here.

Can we verify? Would a free be appropriate regardless?

Reply:
The ownership of `si->pctx` and responsibility for freeing can be tricky.
Here we have obviously one case, where an `si->pctx` is not owned,
and if there are other cases, where that can be the case, or if this
function is called recursively, anything can happen.
As long as there is no test case, where this happens I would not
want to touch this, because currently it is only a theoretically possible
memory leak that nobody has ever been able to reproduce, BUT
if your assumption is wrong, it is a crash that can happen sometimes.
So we don't know how to test the code, and verification is impossible
unless there is some kind of test case for the scenario.